python/ray/tune/experiment/trial.py

```diff
@@ -1093,6 +1093,7 @@ def on_checkpoint(self, checkpoint: Union[_TrackedCheckpoint, _TrainingResult]):
             self.storage.current_checkpoint_index += 1
         else:
             self.run_metadata.checkpoint_manager.on_checkpoint(checkpoint)
+        self.invalidate_json_state()
```

Comment:
`self.storage` is not part of `run_metadata`, so it's not enough to just invalidate the `run_metadata` cache. @krfricke